src/web/components/ClickToView.tsx

```diff
@@ -89,74 +113,115 @@ const Headline = ({
+	handleClick,
+}: {
+	role: RoleType;
+	handleClick: Function;
+}) => {
+	return (
+		<div>
+			<Button
+				priority="primary"
+				size={roleButtonSize(role)}
+				icon={<SvgCheckmark />}
+				iconSide="left"
+				onClick={() => handleClick()}
+			>
+				{roleButtonText(role)}
+			</Button>
+		</div>
+	);
+};
 
-export const ClickToView = ({ children, width, height, onAccept }: Props) => {
+export const ClickToView = ({
+	children,
+	role,
+	onAccept,
+	isTracking,
+	source,
+	sourceDomain,
+}: Props) => {
 	const [showOverlay, setShowOverlay] = useState<boolean>(true);
 
 	const handleClick = () => {
 		setShowOverlay(false);
-		onAccept();
+		if (onAccept) {
+			setTimeout(() => onAccept());
+		}
 	};
 
-	if (showOverlay) {
+	let headlineText;
+	let body;
+
+	const roleWithDefault: RoleType = role || 'inline';
+
+	if (isTracking && showOverlay) {
+		if (source) {
+			headlineText = `Allow ${source} content?`;
+			body = (
+				<div>
+					<p>
+						This article includes content provided by {source}. We
+						ask for your permission before anything is loaded, as
+						they may be using cookies and other technologies.
+					</p>
+					<p>
+						To view this content, click &apos;Allow and
+						continue&apos;.
+					</p>
+				</div>
+			);
+		} else {
+			headlineText = 'Allow content provided by a third party?';
+			body = (
+				<div>
+					<p>
+						This article includes content hosted on {sourceDomain}.
+						We ask for your permission before anything is loaded, as
+						the provider may be using cookies and other
+						technologies.
+					</p>
+					<p>
+						To view this content, click &apos;Allow and
+						continue&apos;.
+					</p>
+				</div>
+			);
+		}
+
 		return (
-			<Container width={width} height={height}>
-				<Outer>
-					<Inner>
-						<Top>
-							<Headline width={width}>
-								Allow content provided by a third party?
-							</Headline>
-							<Body width={width}>
-								<p>
-									This article includes content hosted on
-									other.com. We ask for your permission before
-									anything is loaded, as the provider may be
-									using cookies and other technologies.
-								</p>
-								<p>
-									To view this content, click &apos;Allow and
-									continue&apos;.
-								</p>
-							</Body>
-						</Top>
-						<Bottom>
-							<Button
-								priority="primary"
-								size={width > 300 ? 'small' : 'xsmall'}
-								icon={<SvgCheckmark />}
-								iconSide="left"
-								onClick={() => handleClick()}
-							>
-								Allow and continue
-							</Button>
-						</Bottom>
-					</Inner>
-					<Lines />
-				</Outer>
+			<Container>
+				<Headline role={roleWithDefault}>{headlineText}</Headline>
+				<Body role={roleWithDefault}>{body}</Body>
+				<AcceptButton
+					role={roleWithDefault}
+					handleClick={handleClick}
+				/>
 			</Container>
 		);
 	}
 
-	return (
-		<Container width={width} height={height}>
-			{children}
-		</Container>
-	);
+	return <div>{children}</div>;
```

Comment:
We can remove these divs (or use fragments if needed)

Reply:
👍